lib/iris/analysis/__init__.py

```diff
@@ -1597,7 +1601,13 @@ def regridder(self, src_grid, target_grid):
                 `callable(cube)`
 
             where `cube` is a cube with the same grid as `src_grid`
-            which is to be regridded to the `target_grid`.
+            that is to be regridded to the `target_grid`.
+            Thus for the callable returned by
+            `Linear().regridder(src_grid, target_grid)`, cube must
+            be on the same grid as src_grid.
+
+            This callable would typically be used by
+            :class:`iris.cube.Cube.regrid`.
```

Comment:
:+1: for greater clarity in the description of the `regridder()` method and its resulting callable. However, I don't think the "Thus for the callable ..." statement helps. And the "This callable would typically ..." statement risks making matters worse. (e.g. Am I supposed to pass the result of Linear().regridder(..., ...) to cube.regrid()?)

How about:
 - Leaving the "Returns:" section as it was prior to this PR (other than the "which" -> "that" substitution).
 - Adding more detail above the "Args:" section to explain that the first port of call is to use Cube.regrid() and not use this method at all. But when doing repeated regridding operations and cube.regrid() is too inefficient then this section can demonstrate an example usage pattern. As a rough and ready example:
```python
regridder = Linear().regridder(src_grid, target_grid)
for cube in my_long_list_of_cubes_on_the_src_grid:
    cube_on_target_grid = regridder(cube)
    # Do stuff with cube_on_target_grid ...
```